### DISCUSSION TITLE
Using current Virtual user

### DISCUSSION THREAD

**Discussion by efabisze** [artilleryio/artillery#1541] [Artillery]
Is there any way to have access to the current virtual user?  

For example, i am trying to write a custom JS to have a dynamic json :           

`    function setNoteJson(requestParams, context, ee, next)
    {
        requestParams.body = JSON.stringify({
            username: "Note for Virtual User : " + {{context.vuser}},
        });

        return next();
    }
`
Basically wanting the json to have a dynamic note depending on the current virtual user.

**Comment by hassy** [comment]
Each VU has a unique identifier (a [UUID v4](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)) string), which is accessible through the `$uuid` variable in YAML, or via `context.vars.$uuid` in custom Javascript code. Is that what you were looking for @efabisze?

**Reply by efabisze** [reply]
@hassy Exactly what i was looking for, Thank you very much!
